src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts

```diff
@@ -99,7 +108,27 @@ export class AslVisualization {
         )
 
         const sendUpdateMessage = async (updatedTextDocument: vscode.TextDocument) => {
-            const isValid = await isDocumentValid(updatedTextDocument)
+            const isYAML = updatedTextDocument.languageId === 'yaml'
+            const text = updatedTextDocument.getText()
+            let stateMachineData = text
```

Comment:
does this stay empty when isYAML is false? Looks like that might fail isDocumentValid check

Reply:
State machine data is a text of `updatedTextDocumen` - there is an assignment there. If it is a yaml file this variable get reassigned with a new value.